Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -29,6 +29,7 @@ public class FreezeOnMatchCustomization : ICustomization
         /// <exception cref="ArgumentNullException">
         /// <paramref name="targetType"/> is <see langword="null"/>.
         /// </exception>
+        [Obsolete("Please use FreezeOnMatchCustomization(object, IRequestSpecification) instead.")]
```

Comment:
I was thinking about such a scenario, but if we use the `object request` parameter, the constructor overload that does not require the explicit `IRequestSpecification` [may fail in some cases](https://github.com/AutoFixture/AutoFixture/pull/697#discussion_r75891426).